notebooks/32-builtin-types-nb.md

```diff
@@ -645,6 +645,32 @@ for (let key in bond) {
 }
 ```
 
+````{admonition} practice
+:class: seealso
+
+Remember that it is not obvious when to use `in` or `of`. Try to guess what the following code returns:
+````
+
+```{code-cell}
+tab = ["A", "B", "C"]
+obj = {x: "A", y: "B", z: "C"}
+tabAsObj = {"1": "A", "2": "B", "3": "C"}
+
+for (let i in tab) 
+    console.log(`value=${i} type=${typeof i}`)
+
+for (let i of tab)
+    console.log(`value=${i} type=${typeof i}`)
+
+//and the try it out with obj et tabAsObj
+```
+
+````{admonition} practice
+:class: seealso
+
+Javascript seems sometimes... unnatural. Check [destroyallsoftware talks on that](https://javascript.info/number)https://www.destroyallsoftware.com/talks/wat), at 1 min. 21.
```

Comment:
the link feels broken, there are leftovers of javascript.info

Reply:
rhaa, I will modify the PR.